web/src/views/InvestigationDetail.vue

```diff
@@ -486,8 +494,57 @@ export default defineComponent({
       clampBoundsX(false);
     }
 
+    const similarNuclei: Ref<Array<DetectedStructureWithColor>> = ref([]);
+    const axios = inject('axios') as AxiosInstance;
+    function colorSimilarNuclei(nuclei: Array<DetectedStructure>) {
+      const colors = ['#FF0000', '#00FF00', '#0000FF', '#FFFF00', '#FF00FF', '#00FFFF'];
+      const nucleiWithColor: Array<DetectedStructureWithColor> = [];
+      for (let i = 0; i < nuclei.length; i += 1) {
+        const nucleus = nuclei[i];
+        const nucleusWithColor = { ...nucleus, color: colors[i] };
+        nucleiWithColor.push(nucleusWithColor);
+      }
+      return nucleiWithColor;
+    }
+    function setupSimilarNuclei() {
+      const layer = createLayer(
+        'feature',
+        { features: ['marker'] },
+      ) as ReturnType<typeof useGeoJSLayer>;
+      layer.addGeoEventHandler(geoEvents.mouseclick, (event: MouseClickEvent['mouse']) => {
+        const { x, y } = event.geo;
+        if (!rootDataset.value) {
+          return;
+        }
+        const url = `/similar-nuclei/${rootDataset.value.id}/${x}/${y}/`;
+        axios.get(url).then((resp) => {
```

Comment:
Consider using async/await here.

Reply:
Awesome. More readable and less nesting. Resolved in a5609958c0259ed93e214de0e88b0f7ae2953355.